server/src/test/java/org/elasticsearch/common/geo/EdgeTreeTests.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.geo;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.geo.builders.ShapeBuilder;
+import org.elasticsearch.geo.geometry.Polygon;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.geo.RandomShapeGenerator;
+import org.locationtech.spatial4j.shape.Rectangle;
+
+import java.io.IOException;
+
+public class EdgeTreeTests extends ESTestCase {
+
+    public void testRectangleShape() throws IOException {
+        for (int i = 0; i < 1000; i++) {
+            int minX = randomIntBetween(-180, 170);
```

Comment:
Can we leverage lucene's `GeoTestUtil`? There are test methods that tend to create adversarial rectangles that either cross dateline (`nextBox`) or not (`nextBoxNotCrossingDateline`)?